### ISSUE TITLE
(SearchBox) Support OnChange to fire only on enter or losing focus

### ISSUE THREAD

**Issue by scottdurow** [microsoft/powercat-creator-kit#237]
The standard text box only fires the `OnChange` event when the user presses enter or moves focus away from the control.
This is required to allow fast searching without tabbing away to another button to execute the search.

Currently, the `OnChange` event fires with every key press.

This is related to microsoft/powercat-code-components#82

**Comment by bronsonacoutts** [comment]
I agree, I would be much nicer to have this aligned with the existing text input so that OnChange() can be set to Select() a button i.e. the search button for faster searching.
